### PR TITLE
[DI] Fix dumping expressions accessing single-use private services

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 4.2
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #29403
| License       | MIT
| Doc PR        | n/a

Introduced in #27268, see fixed ticket

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php [hunk 1]
```diff
@@ -1650,6 +1650,8 @@ private function getServiceCall(string $id, Reference $reference = null): string
             }
             if ($definition->isShared() && !isset($this->singleUsePrivateIds[$id])) {
                 $code = sprintf('($this->%s[\'%s\'] ?? %s)', $definition->isPublic() ? 'services' : 'privates', $id, $code);
+            } elseif ($trivial) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: Will the variable be always defined?

Comment (chalasr) [reply src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: good catch, it'll be now :)

**Comment by nicolas-grekas** [comment]
I changed L602 recently and I think it should be reverted as it's a very local workaround. Can you have a look please?

**Comment by chalasr** [comment]
@nicolas-grekas done

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @chalasr.

**Pull request merged**